os/win/window.h

```diff
@@ -138,6 +139,8 @@ namespace os {
     gfx::Rect m_restoredFrame;
 #endif
 
+    DragTargetAdapter m_dragTargetAdapter = DragTargetAdapter(this);
```

Comment:
One thing I don't like about this (and calling RegisterDragDrop) for every window is that there are windows (like menu popups/tooltips) that don't require this drag-and-drop logic.

Reply:
Could we convert this to a pointer and initialize the drag-and-drop logic if the m_dragTarget is set?

Reply:
I think so, will try it.

Reply:
Done